core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -2022,6 +2092,8 @@ oppia.factory('explorationWarningsService', [
         }
       });
 
+      _getStatesWithIncorrectSolution();
```

Comment:
I think we have a problem here. Because the computation here is async you're gonna end up with a race condition since you don't have control over when _updateWarningsList() gets called next.

See my comment in StateSolutionHelperService for a suggestion on how to address this.